Can you change the variable name to nodeCapacity to conform with naming standards. Otherwise looks good.

quadtree.go:71:2: don't use underscores in Go names; struct field node_capacity should be nodeCapacity
quadtree.go:81:25: don't use underscores in Go names; func parameter node_capacity should be nodeCapacity

--- a/quadtree.go
+++ b/quadtree.go
@@ -68,7 +68,7 @@ func (aabb *AABB) IntersectsAABB(other *AABB) bool {
 type QuadTree struct {
 	boundary      AABB
 	points        []*XY
-	node_capacity int
+	nodeCapacity int
 	northWest *QuadTree
 	northEast *QuadTree
 	southWest *QuadTree
@@ -76,14 +76,14 @@ type QuadTree struct {
 }
 
 // New creates a new quadtree node that is bounded by boundary and contains
-// node_capacity points.
-// node_capacity is the maximum number of points allowed in a quadtree node
-func New(boundary AABB, node_capacity int) *QuadTree {
-	points := make([]*XY, 0, node_capacity)
+// nodeCapacity points.
+// nodeCapacity is the maximum number of points allowed in a quadtree node
+func New(boundary AABB, nodeCapacity int) *QuadTree {
+	points := make([]*XY, 0, nodeCapacity)
 	qt := &QuadTree{
 		boundary: boundary,
 		points: points,
-		node_capacity: node_capacity,
+		nodeCapacity: nodeCapacity,
 	}
 	return qt
 }
@@ -135,22 +135,22 @@ func (qt *QuadTree) subDivide() {
 	box := AABB{
 		XY{qt.boundary.center.X - qt.boundary.halfDim.X/2, qt.boundary.center.Y + qt.boundary.halfDim.Y/2},
 		XY{qt.boundary.halfDim.X / 2, qt.boundary.halfDim.Y / 2}}
-	qt.northWest = New(box, qt.node_capacity)
+	qt.northWest = New(box, qt.nodeCapacity)
 
 	box = AABB{
 		XY{qt.boundary.center.X + qt.boundary.halfDim.X/2, qt.boundary.center.Y + qt.boundary.halfDim.Y/2},
 		XY{qt.boundary.halfDim.X / 2, qt.boundary.halfDim.Y / 2}}
-	qt.northEast = New(box, qt.node_capacity)
+	qt.northEast = New(box, qt.nodeCapacity)
 
 	box = AABB{
 		XY{qt.boundary.center.X - qt.boundary.halfDim.X/2, qt.boundary.center.Y - qt.boundary.halfDim.Y/2},
 		XY{qt.boundary.halfDim.X / 2, qt.boundary.halfDim.Y / 2}}
-	qt.southWest = New(box, qt.node_capacity)
+	qt.southWest = New(box, qt.nodeCapacity)
 
 	box = AABB{
 		XY{qt.boundary.center.X + qt.boundary.halfDim.X/2, qt.boundary.center.Y - qt.boundary.halfDim.Y/2},
 		XY{qt.boundary.halfDim.X / 2, qt.boundary.halfDim.Y / 2}}
-	qt.southEast = New(box, qt.node_capacity)
+	qt.southEast = New(box, qt.nodeCapacity)
 
 	for _, v := range qt.points {
 		if qt.northWest.Insert(v) {
@@ -170,7 +170,7 @@ func (qt *QuadTree) subDivide() {
 }
 
 func (qt *QuadTree) SearchArea(a *AABB) []*XY {
-	results := make([]*XY, 0, qt.node_capacity)
+	results := make([]*XY, 0, qt.nodeCapacity)
 
 	if !qt.boundary.IntersectsAABB(a) {
 		return results

